mesonbuild/interpreter/interpreter.py

```diff
@@ -867,7 +867,8 @@ def disabled_subproject(self, subp_name: str, disabled_feature: T.Optional[str]
         self.subprojects[subp_name] = sub
         return sub
 
-    def do_subproject(self, subp_name: str, kwargs: kwtypes.DoSubproject, force_method: T.Optional[wrap.Method] = None) -> SubprojectHolder:
+    def do_subproject(self, subp_name: str, kwargs: kwtypes.DoSubproject, force_method: T.Optional[wrap.Method] = None,
+                      extra_default_options: T.Optional[T.Dict[str, str]] = None) -> SubprojectHolder:
```

Comment:
Shouldn't this be `None | dict[str, ElementaryOptionValues]`?

Reply:
Better, yes.